services/catarse/app/api/catarse/v2/billing/payments_api.rb

```diff
@@ -24,6 +24,16 @@ class PaymentsAPI < Grape::API
           # TODO: Payment in installments
           payment_params = declared(params, include_missing: false)[:payment]
 
+          new_address = ::Shared::Addresses::GenerateCopy.result(
+            billing_address_id: payment_params['billing_address_id'],
+            shipping_address_id: payment_params['shipping_address_id']
+          )
+
+          payment_params[:billing_address_id] = new_address.copy_billing_address_id
+          if payment_params['shipping_address_id'].present?
+            payment_params[:shipping_address_id] = new_address.copy_shipping_address_id
+          end
+
```

Comment:
**Problema**
Não é bom tu chamar tuas actions dentro de uma requisição da API, senão daqui a pouco as rotas da API estarão cheias de regra de negócio, chamando várias actions. Fora que torna difícil o reaproveitamento desse código e também a action `Payments::Create` se for chamada diretamente, irá apontar para os endereços que foram passados e não fará uma duplicação deles.